velox/functions/sparksql/Rand.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions {
+
+template <typename T>
+struct RandFunction {
+  static constexpr bool is_deterministic = false;
+
+  FOLLY_ALWAYS_INLINE void call(double& result) {
+    result = folly::Random::randDouble01();
+  }
+
+  FOLLY_ALWAYS_INLINE void callNullable(double& result, const int64_t* seed) {
+    std::mt19937 rng;
+    // For null input, a fixed seed is always used.
+    if (!seed) {
+      rng.seed(0);
+    } else {
+      rng.seed(*seed);
```

Comment:
What is a common use of 'seed'? Is it constant for all rows? Won't this result in all rows producing same "rand" value?